spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -4,19 +4,35 @@ import com.datastax.driver.core._
 import com.datastax.spark.connector.types.ColumnType
 import org.apache.spark.Logging
 
+/**
+ * Class for binding row-like objects into prepared statements. prefixVals
+ * is used for binding constant values into each bound statement. This supports parametrized
+ * .where clauses in [[com.datastax.spark.connector.rdd.CassandraJoinRDD]]
+ */
 private[connector] class BoundStatementBuilder[T](
     val rowWriter: RowWriter[T],
     val preparedStmt: PreparedStatement,
-    val protocolVersion: ProtocolVersion) extends Logging {
+    val protocolVersion: ProtocolVersion,
+    val prefixVals: Seq[Any] = Seq.empty) extends Logging {
 
   private val columnNames = rowWriter.columnNames.toIndexedSeq
   private val columnTypes = columnNames.map(preparedStmt.getVariables.getType)
   private val converters = columnTypes.map(ColumnType.converterToCassandra(_)(protocolVersion))
   private val buffer = Array.ofDim[Any](columnNames.size)
 
+  private val prefixTypes = (0 until prefixVals.length).map(preparedStmt.getVariables.getType)
+  private val prefixConverted = for {
```

Comment:
I assume the current implementation was done to maximize performance. I find the following easier to read though probably less performant. Using a view is debatable (just depends on how big prefixVals is). Since prefixVals is normally going to be small, it should probably not be used.

```
  private val prefixConverted = prefixVals.view.zipWithIndex.map { case (prefixVal, i) =>
    val prefixType = preparedStmt.getVariables.getType(i)
    ColumnType.converterToCassandra(prefixType)(protocolVersion).convert(prefixVal)
  }
```